File: src/main/java/ink/ptms/adyeshach/common/util/BukkitUtils.java
package ink.ptms.adyeshach.common.util;

import org.bukkit.DyeColor;
import org.bukkit.entity.*;
import org.bukkit.inventory.EquipmentSlot;

/**
 * 求你们不要改这个类，用 java 写是有原因的。
 *
 * @author sky
 * @since 2020-08-05 16:53
 */
public class BukkitUtils {

    public static DyeColor[] valuesDyeColor() {
        return DyeColor.values();
    }

    public static Cat.Type[] valuesCatType() {
        return Cat.Type.values();
    }

    public static Fox.Type[] valuesFoxType() {
        return Fox.Type.values();
    }

    public static Llama.Color[] valuesLlamaColor() {
        return Llama.Color.values();
    }

    public static Rabbit.Type[] valuesRabbitType() {
        return Rabbit.Type.values();
    }

    public static Parrot.Variant[] valuesParrotVariant() {
        return Parrot.Variant.values();
    }

    public static Spellcaster.Spell[] valuesIllagerSpell() {
        return Spellcaster.Spell.values();
    }

    public static EquipmentSlot[] valuesEquipmentSlot() {
        return EquipmentSlot.values();
    }
}

File: src/main/kotlin/ink/ptms/adyeshach/common/util/BukkitUtils.kt
package ink.ptms.adyeshach.common.util

import org.bukkit.DyeColor
import org.bukkit.entity.*

/**
 * @author Arasple
 * @date 2020/8/19 21:25
 */
object BukkitUtils {

    fun valuesDyeColor(): Array<DyeColor> {
        return DyeColor.values()
    }

    @Suppress("DEPRECATION")
    fun valuesHorseVariant(): Array<Horse.Variant> {
        return Horse.Variant.values()
    }

    fun valuesHorseColor(): Array<Horse.Color> {
        return Horse.Color.values()
    }

    fun valuesHorseStyle(): Array<Horse.Style> {
        return Horse.Style.values()
    }

    fun valuesCatType(): Array<Cat.Type> {
        return Cat.Type.values()
    }

    fun valuesLlamaColor(): Array<Llama.Color> {
        return Llama.Color.values()
    }

    fun valuesRabbitType(): Array<Rabbit.Type> {
        return Rabbit.Type.values()
    }

    fun valuesParrotVariant(): Array<Parrot.Variant> {
        return Parrot.Variant.values()
    }

    fun valuesIllagerSpell(): Array<Spellcaster.Spell> {
        return Spellcaster.Spell.values()
    }

}